src/uu/env/src/env.rs

```diff
@@ -165,193 +166,362 @@ pub fn uu_app() -> Command {
+             *
+             * POSIX.1-2017 doesn't seem to mention what to do if the string is malformed (at least
+             * not in "Chapter 8, Environment Variables" or in the definition for environ and various
+             * exec*'s or in the description of env in the "Shell & Utilities" volume).
+             *
+             * It also doesn't specify any checks for putenv before modifying the environ variable, which
+             * is likely why glibc doesn't do so. However, the first set_var argument cannot point to
+             * an empty string or a string containing '='.
+             *
+             * There is no benefit in replicating GNU's env behavior, since it will only modify the
+             * environment in weird ways
+             */
+
+            if name.is_empty() {
+                show_warning!("no name specified for value {}", val.quote());
+                continue;
+            }
+            env::set_var(name, val);
+        }
+
+        if opts.program.is_empty() {
+            // no program provided, so just dump all env vars to stdout
+            print_env(opts.line_ending);
+        } else {
+            // we need to execute a command
+            let prog = Cow::from(opts.program[0]);
+            let args = &opts.program[1..];
+
+            if do_debug_printing {
+                eprintln!("executable: {}", prog.quote());
+                for (i, arg) in args.iter().enumerate() {
+                    eprintln!("arg[{}]: {}", i, arg.quote());
                 }
-                #[cfg(not(unix))]
-                return Err(exit.code().unwrap().into());
             }
-            Err(ref err) if err.kind() == io::ErrorKind::NotFound => return Err(127.into()),
-            Err(_) => return Err(126.into()),
-            Ok(_) => (),
+
+            /*
+             * On Unix-like systems Command::status either ends up calling either fork or posix_spawnp
+             * (which ends up calling clone). Keep using the current process would be ideal, but the
+             * standard library contains many checks and fail-safes to ensure the process ends up being
+             * created. This is much simpler than dealing with the hassles of calling execvp directly.
+             */
+            match process::Command::new(&*prog).args(args).status() {
+                Ok(exit) if !exit.success() => {
+                    #[cfg(unix)]
+                    if let Some(exit_code) = exit.code() {
+                        return Err(exit_code.into());
+                    } else {
+                        // `exit.code()` returns `None` on Unix when the process is terminated by a signal.
+                        // See std::os::unix::process::ExitStatusExt for more information. This prints out
+                        // the interrupted process and the signal it received.
+                        let signal_code = exit.signal().unwrap();
```

Comment:
should we handle the potential errors ? (same question for the next line)

Reply:
this is moved code.
I validated that the `unwraps` are save. So I think there is nothing to address.